jrnl/journals/FolderJournal.py

```diff
@@ -143,10 +147,14 @@ def _get_month_folders(path: pathlib.Path) -> list[pathlib.Path]:
         return
 
     @staticmethod
-    def _get_day_files(path: pathlib.Path) -> list[str]:
-        for child in path.glob(DAY_PATTERN):
+    def _get_day_files(self, extension, path: pathlib.Path) -> list[str]:
+        EXTENSION = extension
+        DAY_PATTERN = "[0-3][0-9]." + EXTENSION
+        for child in path.iterdir():
```

Comment:
I would prefer to keep the glob method rather than using iterdir and regex, for performance reasons. We've seen issues in the past where unpredictable things can happen for users when pointing jrnl to a large directory, and iterdir has been the culprit since it iterates through all files instead of just the expected files.

We can still use the glob approach here by adding the configured extension the day pattern at the start of this method, though.